src/lib.rs

```diff
@@ -347,27 +337,27 @@ pub struct Sleeper<T> {
     /// Current Task / Thread to be woken up.
     pub sleeper: Arc<T>,
     /// mpsc Sender used to register Task / Thread.
-    pub sender: mpsc::Sender<Arc<T>>,
+    pub sender: lockfree::channel::mpsc::Sender<Arc<T>>,
 }
 
 impl<T> Waker<T> {
     /// Register all the Tasks / Threads sent for registration.
-    pub fn register_receivers(&self) {
-        for receiver in self.receiver.try_recv() {
-            self.sleepers.borrow_mut().push(receiver);
+    pub fn register_receivers(&mut self) {
```

Comment:
because of the removal of ReffCell, Waker method register_receivers mutates a vector directly and thus needs to be mutable.

This has the effect that the sync broadcast function becomes mutable as well, requiring the sync::Publisher be a mutable object to be used.